twitter/direct_messages.go

```diff
@@ -14,3 +19,119 @@ type DirectMessage struct {
 	SenderScreenName    string    `json:"sender_screen_name"`
 	Text                string    `json:"text"`
 }
+
+// DirectMessageService provides methods for accessing Twitter status API endpoints.
+type DirectMessageService struct {
+	baseSling *sling.Sling
+	sling     *sling.Sling
+}
+
+// newDirectMessageService returns a new StatusService.
+func newDirectMessageService(sling *sling.Sling) *DirectMessageService {
+	return &DirectMessageService{
+		baseSling: sling.New(),
+		sling:     sling.Path("direct_messages/"),
+	}
+}
+
+// DirectMessageShowParams are the parameters for DirectMessageService.Show
+type DirectMessageShowParams struct {
+	ID int64 `url:"id,omitempty"`
+}
+
+// Show returns the requested DirectMessage and all response messages.
+// https://dev.twitter.com/rest/reference/get/direct_messages/show
+func (s *DirectMessageService) Show(id int64) ([]DirectMessage, *http.Response, error) {
+	params := &StatusShowParams{ID: id}
+	dms := new([]DirectMessage)
+	apiError := new(APIError)
+	resp, err := s.sling.New().Get("show.json").QueryStruct(params).Receive(dms, apiError)
+	return *dms, resp, relevantError(err, *apiError)
+}
+
+// DirectMessageGetParams are the parameters for DirectMessageService.Get
+type DirectMessageGetParams struct {
+	SinceID         int64 `url:"since_id,omitempty"`
+	MaxID           int64 `url:"max_id,omitempty"`
+	Count           int   `url:"count,omitempty"`
+	IncludeEntities *bool `url:"include_entities,omitempty"`
+	SkipStatus      *bool `url:"skip_status,omitempty"`
+}
+
+// Get returns the all the direct messages
+// https://dev.twitter.com/rest/reference/get/direct_messages
+func (s *DirectMessageService) Get(params *DirectMessageGetParams) ([]DirectMessage, *http.Response, error) {
+	if params == nil {
```

Comment:
This is done for some other endpoints to merge a required value into the params. There isn't a reason to do it here. Sling's QueryStruct and BodyForm handle nil arguments.